### PR TITLE
API for checking CRA reporters

### PR DESCRIPTION
**Pull request by doelleri**
For cfpb/hmda-pilot#147

### CODE REVIEW

File: test/controllers/isCraReporterSpec.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            .expect(200)
+            .expect('Content-Type', /json/)
+
+                .expect(/"result":true/)
+
+            .end(function (err, res) {
+                done(err);
+            });
+    });
+
```
Comment (LinuxBozo) [inline test/controllers/isCraReporterSpec.js]: What about tests for when there is no respondent id or year? Does it do what is expected?

**Comment by doelleri** [comment]
@LinuxBozo like this?

**Comment by LinuxBozo** [comment]
Actually, on a second look, I realize it's not all that necessary of a test. Because it's a RESTful route with defined params, of course the framework is going to 404 when the route is not defined by the params expected. 
For some reason, I was thinking about tests for what happens when doing a POST/PUT (which we aren't doing), and so therefore you may have empty/undefined req.params.*. By nature of making then params part of the route definition however, req.params.* are always defined when the route matches. Sorry about that.

**Comment by doelleri** [comment]
@LinuxBozo Do you know how force an `err` to test those branches?

**Comment by LinuxBozo** [comment]
@doelleri one surefire way would be to remove the collection so it doesn't exist before the model query happens. However, this could be problematic, unless you recreate the collection and test data you just removed after that one particular test. 
Usually, you would also have to recreate any indexes, since when you drop a collection, indexes defined for that collection get dropped as well, however, because we're using mockgoose which does an in-memory mock, instead of using a real mongo server, and we're not creating indexes as part of the testdata process, there really aren't any indexes anyway.

**Comment by doelleri** [comment]
I found that when there is no collection the callbacks simply never get executed... is it possible to mock the query method on the model to return an err?

**Comment by LinuxBozo** [comment]
@doelleri I don't right off hand no. I'll do some research into it, since we'll need to repeat that pattern.

**Pull request closed without merging**